src/libmongoc/src/mongoc/mongoc-write-command.c

```diff
@@ -1324,6 +1326,14 @@ _mongoc_write_result_merge (mongoc_write_result_t *result,   /* IN */
       result->n_writeConcernErrors++;
    }
 
+   /* We have an error from the server and should append the entire reply to the
+    * result. */
+   if (result->failed || result->n_writeConcernErrors ||
+       _mongoc_error_is_server (&result->error)) {
+      bson_destroy (&result->raw_error_response);
+      bson_copy_to (reply, &result->raw_error_response);
```

Comment:
Instead of storing only the last server error reply, I suggest storing an array of all error server replies. Having all server error replies may be useful to diagnose the cause of errors in a bulk write operation.

Using an array more closely matches the returned `writeConcernErrors` field. `writeConcernErrors` is an array of all `writeConcernError` found from server error replies.

The specification tests do not test this. I suggest adding a test that receives multiple server error replies to test this behavior. Here is an example: https://github.com/kevinAlbs/mongo-c-driver/commit/50d4e9e446caaa2f6b60a8d15973bae1489a2051